src/main/java/de/tum/in/www1/artemis/service/iris/settings/IrisSubSettingsService.java

```diff
@@ -0,0 +1,322 @@
+            IrisSettingsType settingsType) {
+        if (newSettings == null) {
+            if (parentSettings == null) {
+                throw new IllegalArgumentException("Cannot delete the Code Editor settings");
+            }
+            return null;
+        }
+        if (currentSettings == null) {
+            currentSettings = new IrisCodeEditorSubSettings();
+        }
+        if (settingsType == IrisSettingsType.EXERCISE || authCheckService.isAdmin()) {
+            currentSettings.setEnabled(newSettings.isEnabled());
+        }
+        currentSettings.setAllowedModels(selectAllowedModels(currentSettings.getAllowedModels(), newSettings.getAllowedModels()));
+        currentSettings.setPreferredModel(validatePreferredModel(currentSettings.getPreferredModel(), newSettings.getPreferredModel(), currentSettings.getAllowedModels(),
+                parentSettings != null ? parentSettings.getAllowedModels() : null));
+        currentSettings.setChatTemplate(newSettings.getChatTemplate());
+        currentSettings.setProblemStatementGenerationTemplate(newSettings.getProblemStatementGenerationTemplate());
+        currentSettings.setTemplateRepoGenerationTemplate(newSettings.getTemplateRepoGenerationTemplate());
+        currentSettings.setSolutionRepoGenerationTemplate(newSettings.getSolutionRepoGenerationTemplate());
+        currentSettings.setTestRepoGenerationTemplate(newSettings.getTestRepoGenerationTemplate());
+        return currentSettings;
+    }
+
+    /**
+     * Filters the allowed models of a sub settings object.
+     * If the user is an admin, all models are allowed.
+     * Otherwise, only models that are allowed by the parent settings or the current settings are allowed.
+     *
+     * @param allowedModels        The allowed models of the current settings.
+     * @param updatedAllowedModels The allowed models of the updated settings.
+     * @return The filtered allowed models.
+     */
+    private Set<String> selectAllowedModels(Set<String> allowedModels, Set<String> updatedAllowedModels) {
+        return authCheckService.isAdmin() ? updatedAllowedModels : allowedModels;
+    }
+
+    /**
+     * Validates the preferred model of a sub settings object.
+     * If the user is an admin, all models are allowed.
+     * Otherwise, only models that are allowed by the current settings are allowed.
+     *
+     * @param preferredModel      The preferred model of the current settings.
+     * @param newPreferredModel   The preferred model of the updated settings.
+     * @param allowedModels       The allowed models of the current settings.
+     * @param parentAllowedModels The allowed models of the parent settings.
+     * @return The validated preferred model.
+     */
+    private String validatePreferredModel(String preferredModel, String newPreferredModel, Set<String> allowedModels, Set<String> parentAllowedModels) {
+        if (newPreferredModel == null || newPreferredModel.isBlank()) {
```

Comment:
`isBlank` includes `isEmpty`. From the JavaDocs:
> Returns true if the string is empty or contains only [white space](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html#isWhitespace(int)) codepoints, otherwise false.